fetch-utils.js

```diff
@@ -1,31 +1,35 @@
-const SUPABASE_URL = 'https://gxwgjhfyrlwiqakdeamc.supabase.co';
-const SUPABASE_KEY =
-    'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJyb2xlIjoiYW5vbiIsImlhdCI6MTYzNjQxMTMxMiwiZXhwIjoxOTUxOTg3MzEyfQ.PHekiwfLxT73qQsLklp0QFEfNx9NlmkssJFDnlvNIcA';
+const SUPABASE_URL = 'https://nschbtdnjkjkoshunkhk.supabase.co';
+const SUPABASE_KEY = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Im5zY2hidGRuamtqa29zaHVua2hrIiwicm9sZSI6ImFub24iLCJpYXQiOjE2NDQ0MzYwMjAsImV4cCI6MTk2MDAxMjAyMH0.mwm25ysgqkbA7ZMQcYTguKcFC-qRnmiDHgK2PxCROPk';
+    
 
 const client = supabase.createClient(SUPABASE_URL, SUPABASE_KEY);
 
 export async function createTodo(todo) {
-    // create a single incomplete todo with the correct 'todo' property for this user in supabase
+    const resp = await client.from('todos').insert({ todo, complete: false, user_id: client.auth.user().id, })
+        .single();
+    return checkError(resp);
+    
 
-    return checkError(response);
+    
 }
-
 export async function deleteAllTodos() {
+    const resp = await client.from('todos').delete().match({ user_id: client.auth.user().id, }); 
     // delete all todos for this user in supabase
 
-    return checkError(response);
+    return checkError(resp);
 }
 
 export async function getTodos() {
-    // get all todos for this user from supabase
-
-    return checkError(response);
+    const resp = await client.from('todos').select().order('complete').match({ user_id: client.auth.user().id, });
+    return checkError(resp);
 }
 
 export async function completeTodo(id) {
+    const resp = await client.from('todos').update({ complete: true }).match({ user_id: client.auth.user().id, id: id });
+    // console.log(resp);
     // find the and update (set complete to true), the todo that matches the correct id
 
-    return checkError(response);
+    return checkError(resp);
 }
 
 export async function getUser() {
```

Comment:
You can attach this to your return statement within the get user function "&& client.auth.session().user" then you won't have to call the user_id within the createTodo function, as the user is passed back under our checkAuth() function. Everything looks great! Nice work!